### ISSUE TITLE
ROUTES & MARKER: It is not possible to migrate my markers from the old Soundscape to the new

### ISSUE THREAD

**Issue by StevenAbrams** [soundscape-community/soundscape#49]
**Build: 1.0.0 (8)**

**Goal:**
I want to migrate ALL of my old Soundscape Markers from the old version to the new Soundscape Community version

**Repro Steps:**

1. Open the old/previous version of Soundscape
2. Select "Markers & Routes"
3. Select the "Routes" tab
4. Select the "+" button to start creating a new route
5. Provide a name for the new Route e.g., All my Markers
6. Select the "Waypoints" button
7. Add ALL your Markers
8. Select Done
9. Open the new saved Route
10. Select "Share"
11. Share the Route via SMS to yourself
12. Exit old Soundscape
13. Go to SMS
14. Tap on the shared file

**Expected Results:**
There is some way to save the file and then open the file with the new Soundscape Community app and import all of my previous markers

**Actual Results:**
I can perform no actions on the shared file.

**Comment by StevenAbrams** [comment]
@Oliver2213 with the help of @fraziercarr we did some further investigation.

A route shared from the old Soundscape has a .Soundscape extension.

e.g., 
[685FE288-C34C-4B3E-A399-E1FFDA88046A.soundscape.zip](https://github.com/soundscape-community/soundscape/files/12378036/685FE288-C34C-4B3E-A399-E1FFDA88046A.soundscape.zip)

To be able to open such a file with Soundscape Community, the extension needs to be changed to .openscape
e.g., 
[685FE288-C34C-4B3E-A399-E1FFDA88046A.openscape.zip](https://github.com/soundscape-community/soundscape/files/12378040/685FE288-C34C-4B3E-A399-E1FFDA88046A.openscape.zip)

If you now try to open the .openscape file, the Soundscape Community app will open but the following error will appear:

![IMG_3765](https://github.com/soundscape-community/soundscape/assets/6051244/ffbebd1e-dab9-4bbe-9c17-92b4ce7aadea)

**Comment by Oliver2213** [comment]
I've updated some metadata about what filetypes the app imports, and sharing works again for me - can you confirm this in build 11?

**Comment by StevenAbrams** [comment]
@Oliver2213 @steinbro  This still fails for me when trying to share a route from the old Microsoft Soundscape to the new App Store version of Soundscape Community (build 1.0.0 (13))

My repro steps are as follows:

1. Open Microsoft Soundscape
2. Select "Markers & Routes"
3. Select the "Routes" tab
4. Select a route you wish to Share
5. Select the "Share" option - the Share sheet opens with various share options
6. Select Soundscape Community from the app row (or select the "More" option and choose "Soundscape Community"

**Expected:**
The Soundscape Community app opens and the Route is successfully added. All Markers associated with the route are also added to the current list of Markers.

**Actual:**
An error appears saying "Oops! Something went wrong". If I check the Markers & Routes UI, I don't see the route that I wanted to import.

Attached image shows the error:

![image](https://github.com/soundscape-community/soundscape/assets/6051244/015ddb99-1531-44f7-a8bc-ccd6b72be48b)

**Comment by steinbro** [comment]
Awaiting verification of fix in 1.0.1 test build.

**Comment by RDMurray** [comment]
This isn't currently fixed in 1.0.1(build 1) I forgot about this issue. I'll look at it today.

**Comment by RDMurray** [comment]
Is anyone still having trouble importing routes from microsoft soundscape? I have done it a couple of times with no problem.

**Comment by jchudge** [comment]
User feedback

User previously had trouble importing routes from other Soundscape apps, and the only way to fix it was to re-download and install Microsoft Soundscape and then delete it. 
It turns out that you need to download Microsoft Soundscape for a second time in order to import routes properly from Microsoft Soundscape. Now it doesn't matter how many times you download Microsoft Soundscape, if for any reason you have to uninstall and reinstall Soundscape Community, the original bug returns, and you won't be able to share routes until you reinstall and uninstall Microsoft Soundscape. 

The issues can be reproduced without using any other app except Soundscape Community. 
First, make sure that you don't have any Soundscape apps installed on your phone. Then download Soundscape Community and install it. 
After that, create a route and export it to an Email or iMessage. 
Then delete the original route from Soundscape Community and its associated markers just to make sure. 
Finally, share the route from the Email back to Soundscape Community and see if it works.

This can also be tested without Microsoft Soundscape as it is no longer available by using another version of Soundscape to create the route to import into Soundscape Community.
Start by not having any Soundscape apps installed on the phone. 
Then, download and install Soundscape Community. 
Next, download and install an alternative version of Soundscape in order to create a route.
Then, send the route to Soundscape Community to check if the route can get imported. 
This seems to result in the route not being imported into Soundscape Community.

**Comment by JJGatchalian** [comment]
Did today's commit to fix imported routes resolve this particular issue? If so, I could close this once the next version gets released.

**Comment by RDMurray** [comment]
Yes it did, and I'm embarrassed that this bug has existed for so long. Apologies to anyone affected.

**Comment by JJGatchalian** [comment]
After testing the import routes feature, it works fine even without reinstalling MS Soundscape. Thanks. I think it's safe to close this issue.

**Comment by RDMurray** [comment]
forgot to link this from the PR.

Fixed in #287
